dotcms-integration/src/test/java/com/dotmarketing/util/ContentPublishDateUtilTest.java

```diff
@@ -0,0 +1,178 @@
+package com.dotmarketing.util;
+
+import com.dotcms.api.system.event.message.SystemMessageEventUtil;
+import com.dotcms.concurrent.Debouncer;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotmarketing.beans.Identifier;
+import com.liferay.portal.language.LanguageUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import java.util.Calendar;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Unit tests for {@link ContentPublishDateUtil}.
+ * <p>
+ * Verifies behavior related to future publish date checks, system message handling,
+ * and the use of debouncing for user notifications.
+ */
+class ContentPublishDateUtilTest {
+
+    private ContentType contentType;
+    private Identifier identifier;
+    private Debouncer debouncerMock;
+
+    /**
+     * Initializes mocks and injects a mock Debouncer into the utility class before each test.
+     */
+    @BeforeEach
+    void setUp() {
+        contentType = mock(ContentType.class);
+        identifier = mock(Identifier.class);
+        debouncerMock = mock(Debouncer.class);
+        ContentPublishDateUtil.debouncer = debouncerMock;
+    }
```

Comment:
Never been a fan of mocking static code. I would have made the ContentPublishDateUtil a bean with a smaller lifecycle. The problem of mocking static code is that it becomes a source of trouble when used by other tests. 
Here you are overriding the debouncer with a mock, and perhaps this might affect another test using VersionableAPI. I'd rather we write beans with shorter and more manageable life cycles that are less risky
But if you're not willing to make such a change, then we need to restore the original debouncer when the test has finished

Reply:
do we really need to mock all these objects? This is an IT that can make it easy